src/Engine/GameResourceManager.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Engine/GameResourceManager.h"
+#include "Engine/ErrorHandling.h"
+#include "Engine/Localization.h"
+#include "Engine/Graphics/Image.h"
+
+#include "Utility/DataPath.h"
+
+#include "Library/Compression/Compression.h"
+
+void GameResourceManager::openGameResources() {
+    _eventsLodReader = LodReader::open(MakeDataPath("data", "events.lod"), "icons");
+    if (!_eventsLodReader) {
+        Error(localization->GetString(LSTR_PLEASE_REINSTALL), localization->GetString(LSTR_REINSTALL_NECESSARY));
+    }
+}
+
+Blob GameResourceManager::getEventsFile(const std::string &filename) {
+    assert(_eventsLodReader != nullptr);
+    Blob file = _eventsLodReader->read(filename);
+    return uncompressPseudoTexture(file);
+}
+
+Blob GameResourceManager::uncompressPseudoTexture(Blob &input) {
+    const TextureHeader *header = (const TextureHeader*)input.data();
```

Comment:
You're not checking for data size.

Better use the infrastructure that's already in place:

```
MemoryInputStream stream(input.data(), input.size());
TextureHeader header;
deserialize(stream, &header);
```

You might have to invoke `MM_DECLARE_MEMCOPY_SERIALIZABLE` for `TextureHeader` for this to work.

Also, this should avoid potential UB of reading unaligned integers.